server/routes/controllers/authentication.js

```diff
@@ -0,0 +1,68 @@
+const express = require("express");
+const db = require("../../db/models");
+//const { OAuth2Client } = require("google-auth-library");
+const { google } = require("googleapis");
+const User = require("../../db/models/User");
+const auth = require("../../middleware/auth");
+
+const router = new express.Router();
+
+const oAuthClient = new google.auth.OAuth2(
+  process.env.AUTH_CREDENTIALS,
+  process.env.AUTH_SECRET,
+  process.env.AUTH_REDIRECT_PATH
+);
+
+// login with googleAuth
+router.post("/api/authentication/google", async (req, res) => {
+  //code from google auth, email=user email, variant=type of authentication [login vs signup]
+  const { code, email, variant } = req.body;
+  oAuthClient.getToken(code, async (err, token) => {
+    if (err) {
+      console.log(err);
+      res.status(500).send("Authentication error.", err);
+    }
+    const tokens = {
+      id_token: token.id_token,
+      access_token: token.access_token
+    };
+    const id_token = token.id_token;
+    // const access_token = token.access_token;
+
+    if (variant === "signup") {
+      const userInfo = await oAuthClient.verifyIdToken({
+        idToken: tokens.id_token,
+        audience:
+          "294753578980-nbeunl8bovad0pp6t4ve8p5vso2hiahg.apps.googleusercontent.com"
+      });
+
+      const newUser = new User({
+        email: userInfo.payload.email,
+        name: userInfo.payload.name,
+        googleCredentials: id_token
```

Comment:
I think this is the reason the reason causing the token to expire after half an hour. The id_token is not the refresh token (see this [post](https://medium.com/@nilasini/id-token-vs-access-token-17e7dd622084)). I think it is the `code` sent from the client (in the `GoogleLoginButton.js`) that is the refresh token.